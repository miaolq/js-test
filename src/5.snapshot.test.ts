// snapshot将可序列化的对象（object array jsx），生成快照文件存储，后续变动将对比快照文件
// jest -u 更新快照文件 --updateSnapshot  --testNamePattern
// expect(tree).toMatchSnapshot(...);

const user = {
  name: 'li',
  age: 12,
}

test('basic example', () => {
  expect(user).toMatchSnapshot()
})

const date = {
  now: new Date(),
}

test('property matcher', () => {
  expect(date).toMatchSnapshot({
    now: expect.any(Date),
  })
})

// # Property Matchers

// expect(user).toMatchSnapshot({
//   createdAt: expect.any(Date),
//   id: expect.any(Number),
// })
