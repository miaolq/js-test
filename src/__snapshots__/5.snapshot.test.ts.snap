// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic example 1`] = `
Object {
  "age": 12,
  "name": "li",
}
`;

exports[`property matcher 1`] = `
Object {
  "now": Any<Date>,
}
`;
